crates/runtime/src/debug_builtins.rs

```diff
@@ -0,0 +1,40 @@
+//! Debugger helpers.
+
+use crate::instance::InstanceHandle;
+use crate::vmcontext::VMContext;
+use wasmtime_environ::entity::EntityRef;
+use wasmtime_environ::wasm::MemoryIndex;
+
+static mut VMCTX_AND_MEMORY: (*mut VMContext, usize) = (std::ptr::null_mut(), 0);
+
+#[no_mangle]
+#[allow(dead_code, missing_docs)]
+pub unsafe extern "C" fn resolve_vmctx_memory_ptr(p: *const u32) -> *const u8 {
+    let ptr = std::ptr::read(p);
+    assert!(!VMCTX_AND_MEMORY.0.is_null());
```

Comment:
```suggestion
    assert!(!VMCTX_AND_MEMORY.0.is_null(), "must call `__vmctx->set()` before resolving Wasm pointers");
```

So that users get a slightly better message if they get here.